### ISSUE TITLE
Some logs lost when filling-up too fast?

### ISSUE THREAD

**Issue by stormi** [xcp-ng/xcp#218]
I have seen in one occasion a host where `/var/log/daemon.log` was filling very fast, and due to log rotation (I suppose) it would contain only 1h of logs. So did `/var/log/daemon.log.1.gz`. However `/var/log/daemon.log.2.gz` only contained logs from the day before, so several hours of logs were lost.

There may be something to fix in the log rotation configuration to avoid that.

**Comment by stormi** [comment]
Happens with `kern.log` too: a time gap between `kern.log.1` and `kern.log`. Much annoying when you want to check what happened at a given time and the evidence have been deleted...

**Comment by stormi** [comment]
I have trouble reproducing and nothing in logrotate configuration indicates that such thing should happen. Actually, log rotation only happens once a day (cron.daily), except for /var/log/xensource.log which has a special cron job (`xapi-logrotate.cron`).

The situation is different when the server has the old layout, or no /var/log partition for some reason (typically: installation on Dell servers that have a diagnostics partition, see https://github.com/xcp-ng/xcp/issues/149). In that case, a specific agressive log rotation happens so I wouldn't be surprised if it could be reproduced in that context. However, in the original description I indicated that daemon.log.2.gz existed on the server where I saw the issue, and this can't happen with that agressive log rotation...

I'll keep my eyes open for the next time I meet that issue on a host.
